modin/core/storage_formats/pandas/groupby.py

```diff
@@ -0,0 +1,213 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Contains implementations for GroupbyReduce functions."""
+
+import pandas
+
+from modin.utils import hashable, walk_aggregation_dict
+from modin.core.dataframe.algebra import GroupByReduce
+
+
+class GroupbyReduceImpl:
+    """Provide MapReduce implementations for certain groupby aggregations."""
+
+    @classmethod
+    def get_impl(cls, agg_name):
+        """
+        Get MapReduce implementations for the specified `agg_name`.
+
+        Parameters
+        ----------
+        agg_name : hashable
+
+        Returns
+        -------
+        (map_fn: Union[callable, str], reduce_fn: Union[callable, str], default2pandas_fn: callable)
+        """
+        impl = cls._groupby_reduce_impls.get(agg_name)
+        if impl:
+            return impl
+        raise KeyError(f"Have no implementation for {agg_name}.")
+
+    @classmethod
+    def has_impl_for(cls, agg_func):
+        """
+        Check whether the class has MapReduce implementation for the specified `agg_func`.
+
+        Parameters
+        ----------
+        agg_func : hashable or dict
+
+        Returns
+        -------
+        bool
+        """
+        if hashable(agg_func):
+            return agg_func in cls._groupby_reduce_impls
+        if not isinstance(agg_func, dict):
+            return False
+
+        for _, func, _, _ in walk_aggregation_dict(agg_func):
+            if func not in cls._groupby_reduce_impls:
+                return False
+
+        return True
+
+    @classmethod
+    def register_impl(cls, agg_name):
+        """
+        Register a MapReduce implemented query compiler method for the specified groupby aggregation.
+
+        Parameters
+        ----------
+        agg_name : hashable
+
+        Returns
+        -------
+        callable
+            Function that takes query compiler and executes GroupBy aggregation
+            with TreeReduce algorithm.
+        """
+        map_fn, reduce_fn, d2p_fn = cls.get_impl(agg_name)
+        return GroupByReduce.register(map_fn, reduce_fn, default_to_pandas_func=d2p_fn)
+
+    @staticmethod
+    def _build_skew_impl():
```

Comment:
implementations for `skew` and `mean` were copied here from `PandasQueryCompiler`